crates/cxx-qt/src/lib.rs

```diff
@@ -30,6 +31,21 @@ pub use threading::CxxQtThread;
 #[doc(hidden)]
 pub use static_assertions;
 
+/// Errors that can occur from CXX-Qt
+#[derive(Error, Debug)]
+#[non_exhaustive]
+pub enum CxxQtError {
```

Comment:
consider cxx enum

Reply:
This becomes complex and starts needing to expose types publicly and avoiding collisions when there are multiple QObjects in generated bridges etc. So just leave as an integer for now.